go/kbfs/libgit/commit_file.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2019 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libgit
+
+import (
+	"context"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	billy "gopkg.in/src-d/go-billy.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+)
+
+const (
+	commitFileDefaultMaxBufSize = 4 * 1024 * 1024 // 4 MB
+)
+
+type commitFile struct {
+	hash       plumbing.Hash
+	r          *strings.Reader
+	maxBufSize int64
+	mtime      time.Time
+}
```

Comment:
Could you generalize this into a `diffFile`, rather than being focused on outputting the patch for a single commit?

Reply:
Ok done, hopefully what I did is what you were thinking.